src/StringHelper.php

```diff
@@ -577,4 +579,72 @@ static function (string $key) use ($delimiter, $escapeCharacter): string {
             $result
         );
     }
+
+    /**
+     * Strip Unicode whitespace (with property White_Space=yes) or other characters from the beginning and end of a string.
+     * Input string and pattern are treated as UTF-8.
+     *
+     * @see https://en.wikipedia.org/wiki/Whitespace_character#Unicode
+     * @see https://www.php.net/manual/function.preg-replace
+     *
+     * @param string|string[] $string The string or an array with strings.
+     * @param string $pattern PCRE regex pattern to search for, as UTF-8 string. Quote $pattern if it contains special regular expression characters.
+     * @see https://www.php.net/manual/function.preg-quote.php
+     *
+     * @return array|string
```

Comment:
And the last thing is it would be great to add a psalm generic type like on the playground: https://psalm.dev/r/373f973ea0
Would you add it?

Also change the type please
```suggestion
     * @return string[]|string
```

Reply:
`preg_replace` preserves keys in the returned value, so we can pass associative array of strings